modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -88,9 +91,12 @@ private long parseMillis(String date) {
             }
 
             int year = LocalDate.now(ZoneOffset.UTC).getYear();
-            DateFormatter formatter = DateFormatter.forPattern(format)
-                .withLocale(locale)
-                .withZone(zoneId);
+            DateFormatter dateFormatter = DateFormatter.forPattern(format)
+                .withLocale(locale);
```

Comment:
I suppose locale cannot be null? 
the if condition in JavaDateFormatter is reversed compared to JodaDateFormatter which might throw NPE?

Reply:
indeed, cannot be null, fallback is done in the date processor configuration